Packs/Azure-Enrichment-Remediation/Scripts/AzureFindAvailableNSGPriorities/AzureFindAvailableNSGPriorities.py

```diff
@@ -0,0 +1,86 @@
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+
+
+def find_available_priorities(
+    target_rule_priority: int,
+    number_of_available_priorities_to_retrieve: int,
+    list_of_priorities_from_rules: list,
+) -> list:
+
+    if not target_rule_priority:
+        raise ValueError("target_rule_priority not specified.")
+    elif target_rule_priority <= 100:
+        raise ValueError("target_rule_priority must not be 100 or less.")
+    elif target_rule_priority >= 4096:
+        raise ValueError("target_rule_priority must not be 4096 or more.")
+
+    if not number_of_available_priorities_to_retrieve:
+        raise ValueError("number_of_available_priorities_to_retrieve not specified.")
+    elif number_of_available_priorities_to_retrieve > 5 or number_of_available_priorities_to_retrieve <= 0:
+        raise ValueError(
+            "number_of_available_priorities_to_retrieve cannot be 0 or less, or more than 5. Please use a lower number."
+        )
+
+    if not list_of_priorities_from_rules:
+        raise ValueError("list_of_priorities_from_rules not specified.")
+    if isinstance(list_of_priorities_from_rules, int):
+        list_of_priorities_from_rules = [list_of_priorities_from_rules]
+    if not isinstance(list_of_priorities_from_rules, list):
+        raise ValueError("list_of_priorities_from_rules must be a list.")
+    elif len(list_of_priorities_from_rules) > 999:
+        raise ValueError(
+            "list_of_priorities_from_rules does not support list over 999 entries, please reduce the list."
+        )
```

Comment:
Please move all the validations out to different method

Reply:
Done. 👍